src/modules/tools/zprobe/CartGridStrategy.cpp

```diff
@@ -451,24 +546,27 @@ bool CartGridStrategy::doProbe(Gcode *gc)
 void CartGridStrategy::doCompensation(float *target, bool inverse)
 {
     // Adjust print surface height by linear interpolation over the bed_level array.
-    float grid_x = std::max(0.001F, target[X_AXIS] / AUTO_BED_LEVELING_GRID_X);
-    float grid_y = std::max(0.001F, target[Y_AXIS] / AUTO_BED_LEVELING_GRID_Y);
-    int floor_x = floorf(grid_x);
-    int floor_y = floorf(grid_y);
-    float ratio_x = grid_x - floor_x;
-    float ratio_y = grid_y - floor_y;
-    float z1 = grid[(floor_x) + ((floor_y) * grid_size)];
-    float z2 = grid[(floor_x) + ((floor_y + 1) * grid_size)];
-    float z3 = grid[(floor_x + 1) + ((floor_y) * grid_size)];
-    float z4 = grid[(floor_x + 1) + ((floor_y + 1) * grid_size)];
-    float left = (1 - ratio_y) * z1 + ratio_y * z2;
-    float right = (1 - ratio_y) * z3 + ratio_y * z4;
-    float offset = (1 - ratio_x) * left + ratio_x * right;
-
-    if(inverse)
-        target[Z_AXIS] -= offset;
-    else
-        target[Z_AXIS] += offset;
+    if ((std::min(x_start, x_start + x_size) <= target[X_AXIS]) && (target[X_AXIS] <= std::max(x_start, x_start + x_size)) && 
+        (std::min(y_start, y_start + y_size) <= target[Y_AXIS]) && (target[Y_AXIS] <= std::max(y_start, y_start + y_size))) {
```

Comment:
See above for the note on A,B values (x_size and y_size). According to this, there will be a distance between y_start and y_size that isn't probed and yet is calculated for compensation?

Is x_size and y_size (from AB config) absolute or relative parameters to x_start?